src/aiovodafone/api.py

```diff
@@ -282,6 +283,49 @@ async def _get_csrf_token(self, force_update: bool = False) -> None:
         else:
             _LOGGER.warning("Failed to retrieve CSRF token")
 
+    async def _trigger_diagnostic_call(
+        self,
+        check_url: str,
+        data_url: str,
+        payload: dict[str, Any],
+        key: str,
+        retries: int = 15,
+    ) -> dict:
+        """Trigger a specific diagnostic request to the router."""
+        await self._get_csrf_token()
```

Comment:
The CSRF token needs to be invalidated after a call. 

Otherwise, the server just responds with 403 for all subsequent requests (the first one is okay, though).

```suggestion
        await self._get_csrf_token(force_update=True)
```

Reply:
since the csrf seems to only be required for the diagnostics requests, we can also remove this:

```python
        # Retrieve CSRF token
        # await self._get_csrf_token(force_update=True)
```

From the `login` method.